pkg/spoke/controllers/gccontroller/gc_controller.go

```diff
@@ -0,0 +1,148 @@
+	manifestWorkClient workv1client.ManifestWorkInterface
+	manifestWorkLister worklister.ManifestWorkNamespaceLister
+	spokeDynamicClient dynamic.Interface
+}
+
+// NewGarbageCollectionController returns a GarbageCollectionController
+func NewGarbageCollectionController(
+	recorder events.Recorder,
+	spokeDynamicClient dynamic.Interface,
+	manifestWorkClient workv1client.ManifestWorkInterface,
+	manifestWorkInformer workinformer.ManifestWorkInformer,
+	manifestWorkLister worklister.ManifestWorkNamespaceLister) factory.Controller {
+
+	controller := &GarbageCollectionController{
+		manifestWorkClient: manifestWorkClient,
+		manifestWorkLister: manifestWorkLister,
+		spokeDynamicClient: spokeDynamicClient,
+	}
+
+	return factory.New().
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			return accessor.GetName()
+		}, manifestWorkInformer.Informer()).
+		WithSync(controller.sync).ToController("ManifestWorkGarbageCollectionController", recorder)
+}
+
+func (m *GarbageCollectionController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	manifestWorkName := controllerContext.QueueKey()
+	klog.V(4).Infof("Reconciling ManifestWork %q", manifestWorkName)
+
+	manifestWork, err := m.manifestWorkLister.Get(manifestWorkName)
+	if errors.IsNotFound(err) {
+		// work  not found, could have been deleted, do nothing.
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	return m.syncManifestWork(ctx, manifestWork)
+}
+
+func (m *GarbageCollectionController) syncManifestWork(ctx context.Context, originalManifestWork *workapiv1.ManifestWork) error {
+	manifestWork := originalManifestWork.DeepCopy()
+
+	// no work to do if we're deleted
+	if !manifestWork.DeletionTimestamp.IsZero() {
+		return nil
+	}
+
+	// get the latest applied resources from the manifests in resource status
+	var appliedResources []workapiv1.AppliedManifestResourceMeta
+	for _, resourceStatus := range manifestWork.Status.ResourceStatus.Manifests {
+		gvr := schema.GroupVersionResource{Group: resourceStatus.ResourceMeta.Group, Version: resourceStatus.ResourceMeta.Version, Resource: resourceStatus.ResourceMeta.Resource}
+		if len(gvr.Resource) == 0 || len(gvr.Version) == 0 || len(resourceStatus.ResourceMeta.Name) == 0 {
+			continue
+		}
+
+		appliedResources = append(appliedResources, workapiv1.AppliedManifestResourceMeta{
+			Group:     resourceStatus.ResourceMeta.Group,
+			Version:   resourceStatus.ResourceMeta.Version,
+			Resource:  resourceStatus.ResourceMeta.Resource,
+			Namespace: resourceStatus.ResourceMeta.Namespace,
+			Name:      resourceStatus.ResourceMeta.Name,
+		})
+	}
+
+	// no work to do if applied resources are not changed
+	if reflect.DeepEqual(manifestWork.Status.AppliedResources, appliedResources) {
+		return nil
+	}
+
+	// delete applied resources which are no longer tracked by manifest work
+	untracked := findUntrackedResources(manifestWork.Status.AppliedResources, appliedResources)
```

Comment:
It seems to me that we should find untracked by comparing appliedResource with spec.Workload.Manifests?

Reply:
> It seems to me that we should find untracked by comparing appliedResource with spec.Workload.Manifests?

I think status may be correct here so that you're tracking what has actually been created.